### PR TITLE
Replace ounit2 with Alcotest

### PR DESCRIPTION
**Pull request by MisterDA**
Alcotest has better error reporting.

### CODE REVIEW

**Review by reynir — COMMENTED** [review]
Thanks!

A thing I struggled a bit with before is when running tests and they fail it can be useful to keep temporary files / the archive in order to inspect what went wrong, but it didn't seem too easy to do with ounit2. Do you have ideas how we could improve this? Doesn't have to be part of this PR.

File: lib_test/allocate_set_partial_test.ml [hunk 1]
```diff
@@ -39,45 +38,44 @@ module Test(B : BLOCK) = struct
   let kv_rw_write_error =
     Lwt.wrap1 (Result.fold ~ok:Fun.id ~error:(Fmt.kstr failwith "%a" KV_RW.pp_write_error))
 
-  let connect_block test_ctxt =
-    let filename, ch = bracket_tmpfile ~prefix:"tar-allocate-set-partial-test" ~suffix:".tar" test_ctxt in
-    close_out ch;
+  let connect_block switch () =
```
[the remark this answers is not in the document]
Comment (MisterDA) [reply lib_test/allocate_set_partial_test.ml]: Done.

File: lib_test/parse_test.ml [hunk 2]
```diff
@@ -34,7 +33,14 @@ module Unix = struct
     if Sys.win32 then truncate (convert_path `Windows path) else truncate path
 end
 
-let header _test_ctxt =
+let cstruct = Alcotest.testable
+                (fun f x -> Fmt.pf f "%S" (Cstruct.to_string x))
```
[the remark this answers is not in the document]
Comment (MisterDA) [reply lib_test/parse_test.ml]: Thanks!

**Comment by MisterDA** [comment]
> A thing I struggled a bit with before is when running tests and they fail it can be useful to keep temporary files / the archive in order to inspect what went wrong, but it didn't seem too easy to do with ounit2. Do you have ideas how we could improve this? Doesn't have to be part of this PR.

Can we do this if instead of cleaning files in finalizers, we just clean files "normally"? If an assertion fails, the code won't advance until the cleaning part.

I don't understand yet why the Windows tests are failing. They can't remove the tar files behind them.

**Comment by reynir** [comment]
This looks good to me. I can't figure out why the windows CI fails.

**Review by reynir — APPROVED** [review]

**Comment by MisterDA** [comment]
Let me test on Windows locally. I'll report shortly.

**Comment by MisterDA** [comment]
Can't reproduce the new Windows failure locally, but I've fixed the previous ones.

**Comment by reynir** [comment]
I pushed a fix for something which I believe is broken behavior in the test. The documentation for `Tar.Make.Archive.with_next_file` says:

    (** Read the next header, apply the function 'f' to the fd and the header. The function
        should leave the fd positioned immediately after the datablock. Finally the function
        skips past the zero padding to the next header *)

We do not skip anything in the test. It seems somewhat brittle to expect the function to position the reader at the end of the data.

**Comment by MisterDA** [comment]
Must be transient failures, because now it's succeeding! 
Thanks for all the help!

**Pull request merged**